### PR TITLE
Fix bug in momentum-conserving interpolation function with MR

### PR DESCRIPTION
**Pull request by EZoni**
In the process of generalizing the interpolation functions used with momentum-conserving field gathering and mesh refinement, which will be finalized in a separate PR, I noticed that the new general interpolation function, that I implemented in a separate branch, recovered the behavior of all the old interpolation functions, for the case of refinement ratio equal to 2, except for one, namely `warpx_interp_nd_efield_y`.

I believe that this unveiled a bug in the interpolation function `warpx_interp_nd_efield_y`, similarly to how an earlier bug, fixed in #1584, was unveiled.

All the functions under consideration store interpolated values in variables denoted with the name tags `g`, `c`, and `f`, and store the final result in a variable denoted with the name tag `a` (namely, the `aux` fields), as `a = g + (f - c)`.

The variable with the name tag `c` should contain the result from the interpolation from a coarse staggered grid to a fine nodal grid. This would require to adjust the weights `wx, wy, owx, owy` (in 2D) accordingly.

However, in the case of Ey in 2D, the field is fully nodal. Therefore, the weights do not need to be adjusted and they equal those used to compute the variable with the name tag `g`.

The fact that the weights don't differ probably explains why the old implementation grouped the interpolation of the `g` and `c` values together in the `g` variable only, leaving the `c` variable set to zero:
https://github.com/ECP-WarpX/WarpX/blob/6853c6d6f9a1900779ae4c9aae64d5288d3248aa/Source/Parallelization/WarpXComm_K.H#L524-L529

However, by doing so, the negative sign that the `c` variable carries in assembling the final result as `a = g + (f - c)` is lost.

In other words, the `c` value is computed and stored together with the `g` variable, and hence used in the final result with the wrong sign.

@Reviewers Do you agree with my assessment?

If the bug fix is correct, this requires to reset the benchmark of the CI test `momentum-conserving-gather`, which uses both momentum-conserving field gathering and mesh refinement, in 2D.

### CODE REVIEW

**Comment by RemiLehe** [comment]
Great catch! Thanks for this PR @EZoni

**Review by RemiLehe — APPROVED** [review]
